src/components/watchlist/Watchlist.jsx

```diff
@@ -0,0 +1,88 @@
+import * as React from "react";
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import {
+  GetMoviesBasedOnId,
+  GetWatchlist,
+  AddToWatchlist,
+} from "../../API/movies";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import Divider from "@mui/material/Divider";
+import ListItemText from "@mui/material/ListItemText";
+import ListItemAvatar from "@mui/material/ListItemAvatar";
+import Avatar from "@mui/material/Avatar";
+import Typography from "@mui/material/Typography";
+
+const Watchlist = () => {
+  const { id } = useParams();
+  const [watchlist, setWatchlist] = useState(null);
+
+  useEffect(() => {
+    async function FetchData() {
+      const watchlist = await GetWatchlist().then((resp) => {
```

Comment:
either use async await here or in the api file, please remove redundant async awaits everywhere

Reply:
async and awaits are only being used for the create, delete and update requests. For the get request I am using async await in the useEffect only.